examples/api/test/material/material_state/material_state_property.0_test.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/material_state/material_state_property.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  Finder findForegroundColor(Color color) {
+    return find.byWidgetPredicate((Widget widget) {
+      if (widget is! Material) {
+        return false;
+      }
+      final TextStyle? textStyle = widget.textStyle;
+      return textStyle?.color == color;
+    });
+  }
+
+  testWidgets(
+    'The foreground color of the TextButton should be red by default',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const example.MaterialStatePropertyExampleApp(),
+      );
+
+      expect(findForegroundColor(Colors.red), findsOne);
```

Comment:
Using a finder here feels somewhat unintuitive to me.
Maybe a helper function such as `getButtonForegroundColor` and an expect such as `expect(getButtonForegroundColor(), Colors.red)` would be easier to understand?

Reply:
Sure! What do you think of [revamp: Change findForegroundColor into getButtonForegroundColor](https://github.com/flutter/flutter/pull/156719/commits/1670c7fd0c21909c2768985dbc804cddb767d140)?